src/utils/cScreenShake.cpp

```diff
@@ -0,0 +1,61 @@
+#include "utils/cScreenShake.h"
+#include <algorithm>
+#include <cstdlib>
+#include <iostream>
+#include "map/cMapCamera.h"
+#include "utils/RNG.hpp"
+
+cScreenShake::cScreenShake()
+    : m_shakeX(0), m_shakeY(0), m_timer(0)
+{}
+
+void cScreenShake::shake(int duration)
+{
+    m_timer += duration;
+}
+
+void cScreenShake::reduce()
+{
+    if (m_timer > 0)
+        m_timer--;
+}
+
+void cScreenShake::reset()
+{
+    m_shakeX = 0;
+    m_shakeY = 0;
+    m_timer = 0;
+}
+
+int cScreenShake::getX() const
+{
+    return m_shakeX;
+}
+
+int cScreenShake::getY() const
+{
+    return m_shakeY;
+}
+
+int cScreenShake::getTimer() const
+{
+    return m_timer;
+}
+
+void cScreenShake::update(int state, int playingState, cMapCamera *mapCamera)
+{
+    if (m_timer == 0) {
+        m_timer = -1;
+    }
+
+    if (m_timer > 0 && state == playingState) {
+        // the more we get to the 'end' the less we 'throttle'.
```

Comment:
Seeing this - after a long time - I wonder if we can add another variable that means 'intensity'. Ie, so the shaking is more intens at start and reduces after the fix amount of time. The intensity is now `shakiness` where we do some division and whatnot. Perhaps we can rethink this? In a different PR perhaps.